responsemanager/querypreparer.go

```diff
@@ -20,9 +22,10 @@ import (
 )
 
 type queryPreparer struct {
-	requestHooks      RequestHooks
-	responseAssembler ResponseAssembler
-	linkSystem        ipld.LinkSystem
+	requestHooks       RequestHooks
+	responseAssembler  ResponseAssembler
+	linkSystem         ipld.LinkSystem
+	maxLinksPerRequest uint64
```

Comment:
can we be clear/consistent about whether 0 means fail immediately or no limit?
In this file it seems to mean 'no limit', but in the test in traverser_test "errors correctly, no budget" link budget is set to 0, and the traversal then fails due to being out of budget.

Reply:
in means no limit. The traverser test gets a nil value for the budget, meaning no budget, when this value is 0

Reply:
i think what i'm asking for is a comment to that effect :)

Reply:
added